src/addBenchmarkEntry.ts

```diff
@@ -0,0 +1,41 @@
+import { Benchmark } from './extract';
+import * as core from '@actions/core';
+import { BenchmarkSuites } from './write';
+import { normalizeBenchmark } from './normalizeBenchmark';
+
+export function addBenchmarkEntry(
+    benchName: string,
+    benchEntry: Benchmark,
+    entries: BenchmarkSuites,
+    maxItems: number | null,
+): { prevBench: Benchmark | null; normalizedCurrentBench: Benchmark } {
+    let prevBench: Benchmark | null = null;
+    let normalizedCurrentBench: Benchmark = benchEntry;
+
+    // Add benchmark result
+    if (entries[benchName] === undefined) {
+        entries[benchName] = [benchEntry];
+        core.debug(`No suite was found for benchmark '${benchName}' in existing data. Created`);
+    } else {
+        const suites = entries[benchName];
+        // Get last suite which has different commit ID for alert comment
+        for (const e of suites.slice().reverse()) {
```

Comment:
nit: alternatively you could do something like:

```js
prevBench = suits.slice().reverse().find(e => e.commit.id !== benchEntry.commit.id)
```

it is exactly the same but less verbose

Reply:
Yes, I'll go through the code to simplify some of the old, verbose sections in a follow-up PR. I have this planned. This is just a simple function extraction